app/src/main/java/com/example/brsons/LoginActivity.java

```diff
@@ -0,0 +1,93 @@
+package com.example.brsons;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.util.Patterns;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class LoginActivity extends AppCompatActivity {
+
+    private FirebaseAuth firebaseAuth;
+    private EditText editTextEmail;
+    private EditText editTextPassword;
+    private Button buttonLogin;
+
+    ProgressDialog pd;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        editTextEmail = findViewById(R.id.email);
+        editTextPassword = findViewById(R.id.password);
+
+        buttonLogin = findViewById(R.id.login);
+
+        firebaseAuth = FirebaseAuth.getInstance();
+
+        buttonLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //input data
+                String email = editTextEmail.getText().toString();
+                String passw = editTextPassword.getText().toString().trim();
+                if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()){
+                    //invalid email
+                    editTextEmail.setError("Invalid Email");
+                    editTextEmail.setFocusable(true);
+                } else {
+                    //valid email
+                    loginUser(email,passw);
+                }
+
+            }
+        });
+
+        pd = new ProgressDialog(this);
+
+    }
+    private void loginUser(String email, String passw) {
+        pd.setMessage("Logging In...");
+        pd.show();
+        firebaseAuth.signInWithEmailAndPassword(email, passw)
+                .addOnCompleteListener(this,  new OnCompleteListener<AuthResult>() {
```

Comment:
If user is alreadylogeed in login screen should not be shown!

There is an interface called AuthStateListener() use this, it changes the autentication state very time when the user is logged in and if he is already loged in then it executes according to logic

  authStateListener = new FirebaseAuth.AuthStateListener() {
                @Override
                public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
 FirebaseUser user = firebaseAuth.getCurrentUser();
                        if (user != null) {
//user is loged in print a message
}
                }
            };